RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -79,32 +78,53 @@ const BeamSpotOnlineObjects* OnlineBeamSpotESProducer::compareBS(const BeamSpotO
   // Convert timeThreshold_ from hours to microseconds for comparison
   auto limitTime = std::chrono::microseconds((std::chrono::hours)timeThreshold_).count();
 
+  // Logic for MC
+  // Assume that the Creation Time for both payloads is 0
+  // Choose the one with highest sigmaZ provided the Beam Type is correct
+  // Default to Fake otherwise
+  if (bs1->GetCreationTime() == 0 && bs2->GetCreationTime() == 0) {
+    if (bs1->GetSigmaZ() > bs2->GetSigmaZ() && bs1->GetBeamType() == 2) {
+      return bs1;
+    } else if (bs2->GetSigmaZ() >= bs1->GetSigmaZ() && bs2->GetBeamType() == 2) {
+      return bs2;
+    } else {
+      return nullptr;
+    }
+  }
+
   // Logic to choose between the two BeamSpots:
   // 1. If both BS are older than limitTime retun fake BS
   // 2. If only one BS is newer than limitTime return it only if it has
   //    sigmaZ larger than sigmaZthreshold_ and the fit converged (BeamType 2)
   // 3. If both are newer than the limit threshold return
   //    the BS that converged and has larger sigmaZ
   if (diffBStime1 > limitTime && diffBStime2 > limitTime) {
+    edm::LogInfo("OnlineBeamSpotESProducer") << "Defaulting to fake becuase both payloads are too old.";
     return nullptr;
   } else if (diffBStime2 > limitTime) {
     if (bs1->GetSigmaZ() > sigmaZThreshold_ && bs1->GetBeamType() == 2) {
       return bs1;
     } else {
+      edm::LogInfo("OnlineBeamSpotESProducer")
+          << "Defaulting to fake because the legacy Beam Spot is not suitable and HLT one is too old.";
       return nullptr;
     }
   } else if (diffBStime1 > limitTime) {
     if (bs2->GetSigmaZ() > sigmaZThreshold_ && bs2->GetBeamType() == 2) {
       return bs2;
     } else {
+      edm::LogInfo("OnlineBeamSpotESProducer")
+          << "Defaulting to fake because the HLT Beam Spot is not suitable and the legacy one too old.";
       return nullptr;
     }
   } else {
     if (bs1->GetSigmaZ() > bs2->GetSigmaZ() && bs1->GetBeamType() == 2) {
       return bs1;
-    } else if (bs2->GetSigmaZ() > bs1->GetSigmaZ() && bs2->GetBeamType() == 2) {
+    } else if (bs2->GetSigmaZ() >= bs1->GetSigmaZ() && bs2->GetBeamType() == 2) {
```

Comment:
without this, in the unlikely event in which both payloads had the same (valid) sigmaZ, a fake Beam Spot would be returned.